presto-orc/src/main/java/io/prestosql/orc/metadata/DwrfMetadataReader.java

```diff
@@ -218,24 +224,17 @@ private static RowGroupIndex toRowGroupIndex(HiveWriterVersion hiveWriterVersion
         ImmutableList.Builder<Integer> positions = ImmutableList.builder();
         for (int index = 0; index < positionsList.size(); index++) {
             long longPosition = positionsList.get(index);
-            int intPosition = (int) longPosition;
 
-            checkState(intPosition == longPosition, "Expected checkpoint position %s, to be an integer", index);
+            @SuppressWarnings("NumericCastThatLosesPrecision")
+            int intPosition = (int) longPosition;
+            checkArgument(intPosition == longPosition, "Expected checkpoint position [%s] to be an integer", index);
```

Comment:
quote `longPosition` as well?

or remove the explicit check and use `toIntExact` instead?

(same in `OrcMetadataReader.java`)